instrumentation/dubbo-rpc/src/main/java/brave/dubbo/rpc/TracingFilter.java

```diff
@@ -96,7 +98,9 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
       Future<Object> future = rpcContext.getFuture(); // the case on async client invocation
       if (future instanceof FutureAdapter) {
         deferFinish = true;
-        ((FutureAdapter) future).getFuture().setCallback(new FinishSpanCallback(span));
```

Comment:
I'm concerned about this change. Before, we set a callback. This ensures the span is finished in case user never sets a callback. Now, we don't. I think this is dangerous as it could lead to leaked spans. I will push a commit to your branch for review.

Reply:
tests were not running. this is why we didn't notice. I will fix the build. I've already fixed the code on this branch.